### ISSUE TITLE
Preprocess of mask 

### ISSUE THREAD

**Issue by UdonDa** [yingchen001/BAT-Fill#5]
Hi, @yingchen001

Thank you for sharing your nice work!

I have two questions about a preprocess of irregular mask dataset.

1. Did you use [NVIDIA Irregular Mask Dataset: Training Set](https://www.dropbox.com/s/qp8cxqttta4zi70/irregular_mask.zip?dl=0) and [NVIDIA Irregular Mask Dataset: Testing Set](https://www.dropbox.com/s/01dfayns9s0kevy/test_mask.zip?dl=0) in [official page](https://nv-adlr.github.io/publication/partialconv-inpainting), right?

2. In the above page, authors said that "please use random augmentation tricks". However, I cannot find augmentation for original masks in https://github.com/yingchen001/BAT-Fill/blob/master/data/inpaint_dataset.py#L187. In fact, I obtain a image as shown in below.
![1](https://user-images.githubusercontent.com/25411643/156166060-d9802ce5-0a0a-4eb9-929f-c2aa36dc9190.jpg)

Would you tell me these answers?

Best regards.

**Comment by UdonDa** [comment]
![2](https://user-images.githubusercontent.com/25411643/156187077-c50780fd-78e1-4fd3-ba33-91595b288751.jpg)
Actually, when I use a test set, I have no problem.

This question of training mask is not intrinsically related to your work, but could you please tell me?

Thanks

**Comment by yingchen001** [comment]
Hi,

Thanks for your interest.

1. We're using the test set of the irregular masks. 
2. We apply random flip on both image and masks during training, as per our codes:
`if self.opt.phase == 'train' and not if_tran_sample: 
            img = transforms.RandomHorizontalFlip()(img)
            img = transforms.ColorJitter(0.05, 0.05, 0.05, 0.05)(img)
            mask = transforms.RandomHorizontalFlip()(mask)
            mask = mask.filter(ImageFilter.MaxFilter(3))`

Regards.

**Comment by UdonDa** [comment]
Thank you for your reply.

You mean you use the test set for training your model, right?

**Comment by yingchen001** [comment]
Yes, that's correct.

**Comment by KANGXI123456** [comment]
@yingchen001 
您好，非常感谢您的分享，麻烦问一下，您的这个模型是将places2的数据直接作为输入的吗？我训练的时候只有这样一个显示
![2022-03-03 20-48-49 的屏幕截图](https://user-images.githubusercontent.com/87471239/156569768-d1c8ed78-a878-46b4-b194-32ae6b9e88fa.png)
训练完之后只有这样两个文件。
![2022-03-03 20-59-55 的屏幕截图](https://user-images.githubusercontent.com/87471239/156569874-3520034d-a89d-4cd0-aedb-ee82360aa875.png)
是哪里出问题了吗？

**Comment by yingchen001** [comment]
Nope, that's all right. The file records the training iterations for continued training, it's okay to train without it.

**Comment by KANGXI123456** [comment]
非常感谢您的恢复！
我这里又出现了新的问题：
1、这个只能训练一个epoch吗，我训练了1个epoch就停了
![2022-03-04 09-05-26 的屏幕截图](https://user-images.githubusercontent.com/87471239/156679908-17b97a3d-aed9-4172-aaaa-d5e1e554c1a5.png)
2、再次训练的有关于resume的设置吗？我直接点训练没有读入之前训练的数据：
![B15D8E0F77E05AF8797AD7E25801B34A](https://user-images.githubusercontent.com/87471239/156680510-035edd0c-5184-436a-acfa-96ce16d75808.jpg)
而且tran_iter.txt文件是空的：
![2022-03-04 09-16-37 的屏幕截图](https://user-images.githubusercontent.com/87471239/156680675-ef787505-6247-4b9b-a6d2-3fc4b02570bc.png)

**Comment by KANGXI123456** [comment]
非常感谢您的回复！
我这里又出现了新的问题：
1、这个只能训练一个epoch吗，我训练了1个epoch就停了
![2022-03-04 09-05-26 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682016-ffd844bc-dcf7-41dc-944c-7f602605a8f4.png)
2、有关于resume的操作吗？我直接运行的话没有读入原来的数据
![B15D8E0F77E05AF8797AD7E25801B34A](https://user-images.githubusercontent.com/87471239/156682130-10e823e3-0dd6-4b02-8982-7c373eb33eee.jpg)

而且这个tran_iter.txt文件里面是空的

![2022-03-04 09-16-37 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682271-d12371cb-c4b6-4759-b12d-f076e42ab2e0.png)

**Comment by KANGXI123456** [comment]
还有个不懂的地方，这里说的预训练权重是在哪里用到的，在训练过程中好像并没有用到这个
![2022-03-04 09-36-09 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682765-cf557e0c-0c13-4cc8-817d-9f406ad64411.png)

**Comment by yingchen001** [comment]
1. It will load the saved checkpoint if you want to continue training.
2. As per error says, you may need to convert the tensor to CPU by using .cpu()
3. The pretrained VGG model is used for training the upsampler, not the transformer.

Thanks.

**Comment by zborger** [comment]
cluster = torch.from_numpy(np.load('/data/vdc/yingchen.yu/project/GPT_Inpaint/kmeans_centers.npy'))
您好，我遇到一个问题，请问这个kmeans_centers.npy文件是需要自己下载，然后放到对应的路径吗

**Comment by zborger** [comment]
> 非常感谢您的回复！ 我这里又出现了新的问题： 1、这个只能训练一个epoch吗，我训练了1个epoch就停了 ![2022-03-04 09-05-26 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682016-ffd844bc-dcf7-41dc-944c-7f602605a8f4.png) 2、有关于resume的操作吗？我直接运行的话没有读入原来的数据 ![B15D8E0F77E05AF8797AD7E25801B34A](https://user-images.githubusercontent.com/87471239/156682130-10e823e3-0dd6-4b02-8982-7c373eb33eee.jpg)
> 
> 而且这个tran_iter.txt文件里面是空的
> 
> ![2022-03-04 09-16-37 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682271-d12371cb-c4b6-4759-b12d-f076e42ab2e0.png)

你好，我也遇到了这个问题，请问你解决了吗。这些类似tran_iter.txt的文件需要自己先建立才会写入吗？还是会自动创建这个文件？

**Comment by yingchen001** [comment]
> kmeans_centers

yes, please change the abs path to the downloaded kmeans path

**Comment by yingchen001** [comment]
> > 非常感谢您的回复！ 我这里又出现了新的问题： 1、这个只能训练一个epoch吗，我训练了1个epoch就停了 ![2022-03-04 09-05-26 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682016-ffd844bc-dcf7-41dc-944c-7f602605a8f4.png) 2、有关于resume的操作吗？我直接运行的话没有读入原来的数据 ![B15D8E0F77E05AF8797AD7E25801B34A](https://user-images.githubusercontent.com/87471239/156682130-10e823e3-0dd6-4b02-8982-7c373eb33eee.jpg)
> > 而且这个tran_iter.txt文件里面是空的
> > ![2022-03-04 09-16-37 的屏幕截图](https://user-images.githubusercontent.com/87471239/156682271-d12371cb-c4b6-4759-b12d-f076e42ab2e0.png)
> 
> 你好，我也遇到了这个问题，请问你解决了吗。这些类似tran_iter.txt的文件需要自己先建立才会写入吗？还是会自动创建这个文件？

they'll be auto created, it's empty because the training is stopped

**Comment by zhangshilong2000** [comment]
你好，这个模代码是在windows上运行的吗
